Link to StatProfileHTML in Profile docs

oxinabox opened the pull request:
If we are going to link to ProfileView then we need to be fair and do both.
I like StatProfileHTML better,

other option is to delete all reference

oxinabox committed:
Link to StatProfileHTML

KristofferC commented:
Also https://github.com/vchuravy/PProf.jl? :P

ViralBShah commented:
Should we still get this in?

oxinabox commented on doc/src/manual/profile.md:
```diff
@@ -57,7 +57,8 @@ julia> using Profile
 
-To see the profiling results, there is a [graphical browser](https://github.com/timholy/ProfileView.jl)
+To see the profiling results, there are several graphical browsers
+including [ProfileView.jl](https://github.com/timholy/ProfileView.jl) and [https://github.com/tkluck/StatProfilerHTML.jl](StatProfilerHTML)
```
```suggestion
including [ProfileView.jl](https://github.com/timholy/ProfileView.jl), [https://github.com/tkluck/StatProfilerHTML.jl](StatProfilerHTML), and [PProf.jl](https://github.com/vchuravy/PProf.jl).
```

oxinabox commented:
yes, probably. because ProfileView has issues on Mac for GTK sometimes, and also handled very large profiles badly because it makes mega SVGs in Jupyter.

timholy commented on doc/src/manual/profile.md:
```diff
@@ -57,7 +57,8 @@ julia> using Profile
 
-To see the profiling results, there is a [graphical browser](https://github.com/timholy/ProfileView.jl)
+To see the profiling results, there are several graphical browsers
+including [ProfileView.jl](https://github.com/timholy/ProfileView.jl), [https://github.com/tkluck/StatProfilerHTML.jl](StatProfilerHTML), and [PProf.jl](https://github.com/vchuravy/PProf.jl).
```
```suggestion
One "family" of visualizers is based on [FlameGraphs.jl](https://github.com/timholy/FlameGraphs.jl), with each family member providing a different user interface:
- [Juno](https://junolab.org/) is a full IDE with built-in support for profile visualization
- [ProfileView.jl](https://github.com/timholy/ProfileView.jl) is a stand-alone visualizer based on GTK
- [ProfileVega.jl](https://github.com/davidanthoff/ProfileVega.jl) uses VegaLight and integrates well with Jupyter notebooks
- [StatProfilerHTML](https://github.com/tkluck/StatProfilerHTML.jl) produces HTML and presents some additional summaries, and also integrates well with Jupyter notebooks
- [ProfileSVG](https://github.com/timholy/ProfileSVG.jl) renders SVG

An entirely independent approach to profile visualization is [PProf.jl](https://github.com/vchuravy/PProf.jl), which uses the external `pprof` tool.
```

timholy replied:
I wasn't even sure it's worth mentioning ProfileSVG, feel free to delete it if you want.

oxinabox commented:
wow during the time since i wrote this it was utterly outdated i see.
Thanks @timholy

timholy commented on doc/src/manual/profile.md:
````diff
@@ -57,7 +57,15 @@ julia> using Profile
 ```
 
-To see the profiling results, there is a [graphical browser](https://github.com/timholy/ProfileView.jl)
+To see the profiling results, there are several graphical browsers
````
```suggestion
To see the profiling results, there are several graphical browsers.
```

timholy commented on doc/src/manual/profile.md:
```diff
@@ -57,7 +57,15 @@ julia> using Profile
+- [ProfileSVG](https://github.com/timholy/ProfileSVG.jl) renders SVG
+
+An entirely independent approach to profile visualization is [PProf.jl](https://github.com/vchuravy/PProf.jl), which uses the external `pprof` tool.
 available, but here we'll use the text-based display that comes with the standard library:
```
```suggestion

Here, though, we'll use the text-based display that comes with the standard library:
```

timholy reviewed (commented):
Yeah, splitting ProfileView into LeftChildRightSiblingTree, FlameGraphs, ProfileView, and ProfileSVG led to a number of other opportunities.

ViralBShah commented:
Would be nice to rebase, reword and get it merged!

oxinabox committed:
Update doc/src/manual/profile.md
Update doc/src/manual/profile.md

Co-Authored-By: Tim Holy <tim.holy@gmail.com>
Apply suggestions from code review

Co-Authored-By: Tim Holy <tim.holy@gmail.com>
Update doc/src/manual/profile.md

Co-Authored-By: Tim Holy <tim.holy@gmail.com>

oxinabox commented:
Done, rebased and partially squashed

KristofferC commented:
It should be noticed that this is GPL licensed.

timholy commented:
Accidental cross post or something we need to deal with?

KristofferC commented:
Nothing really that has to be dealt with but some people are a bit worried to get "tainted" by accidentally reading GPL code (derived work etc) like with StatProfileHTML. Don't know if it is a problem in practice.

ViralBShah commented:
I think it is reasonable to expect that people will check the license of a piece of software before looking at it.

timholy commented:
Thanks for explaining, I wasn't sure what you were referring to. Good catch. I don't think it matters for me (I don't plan on reading through its source), but it would be nice if GitHub displayed a "license summary" on the home screen.
